package.json

```diff
@@ -11,14 +11,18 @@
     "tsc-build": "tsc",
     "tsc-watch": "tsc --watch",
     "tsc-check": "tsc --noEmit",
-    "start": "npx ts-node --esm src/index.ts"
+    "start": "node --import ./ts-node.register.mjs ./src/index.ts"
```

Comment:
Interesting, wonder why this was needed?

Reply:
Couldn't get ts-node to run on properly on Node >=v20 with native ESM modules.

Apparently a breaking change introduced in v20:

> Custom ESM loader hooks run on dedicated thread
> ESM hooks supplied via loaders (--experimental-loader=foo.mjs) now run in a dedicated thread, isolated from the main thread. This provides a separate scope for loaders and ensures no cross-contamination between loaders and application code.

Reply:
That said, @milomg if you have any suggestions how to clean this up in a better way, I'm all ears 👍